orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -0,0 +1,126 @@
+            For :py:obj:`model_desc['outputs']` entries, the order must match the original PyTorch's output as returned by :py:meth:`torch.nn.Module.forward` method.
+            :py:attr:`shape` is a list of string or integers that describes the shape of the input/output.
+            Each dimension size can be either a string or an int. String means the dimension size is dynamic, while integers mean static dimensions.
+            An empty list implies a scalar.
+            Lastly, :py:attr:`is_loss` is a boolean (default is False) that flags if this output is considered a loss.
+            ORT backend needs to know which output is loss in order to generate back propagation graph.
+            Loss output must be specified when either :py:attr:`loss_fn` is specified or when loss is embedded in the model.
+            Note that only one loss output is supported per model.
+        optimizer_config (.optim._OptimizerConfig): optimizer config.
+            One of :py:class:`.optim.Adam`, :py:class:`.optim.Lamb` or :py:class:`.optim.SGD`.
+        loss_fn (default is None): a PyTorch loss function.
+            It takes two inputs [prediction, label] and output a loss tensor.
+            If provided, :py:attr:`loss_fn` is combined with the PyTorch :py:attr:`model` to form a combined PyTorch model.
+            Inputs to the combined PyTorch model are concatenation of the :py:attr:`model`'s input and :py:attr:`loss_fn`'s label input.
+            Outputs of the combined PyTorch model are concatenation of :py:attr:`loss_fn`'s loss output and :py:attr:`model`'s outputs.
+        options (ORTTrainerOptions, default is None): options for additional features.
+    Example:
+        .. code-block:: python
+            model = ...
+            model_desc = {
+                "inputs": [
+                    ("input_ids", ["batch", "max_seq_len_in_batch"]),
+                    ("attention_mask", ["batch", "max_seq_len_in_batch"]),
+                    ("token_type_ids", ["batch", "max_seq_len_in_batch"]),
+                    ("masked_lm_labels", ["batch", "max_seq_len_in_batch"]),
+                    ("next_sentence_label", ["batch", 1])
+                ],
+                "outputs": [
+                    ("loss", [], True),
+                ],
+            }
+            optim_config = optim.Lamb(param_groups = [ { 'params' : ['model_param0'], 'alpha' : 0.8, 'beta' : 0.7},
+                                                       { 'params' : ['model_param1' , 'model_param_2'], 'alpha' : 0.0}
+                                                     ],
+                                      alpha=0.9, beta=0.999)
+            ort_trainer = ORTTrainer(model, model_desc, optim_config)
+    """
+
+    def __init__(self, model, model_desc, optim_config, loss_fn=None, options=None):
+        # Basic validation
+        assert model is not None, "'model' is required and must be either a 'torch.nn.Module' or ONNX model"
```

Comment:
```suggestion
        assert model is not None, "'model' is required and must be either a 'torch.nn.Module' or 'onnx.ModelProto'"
```

Reply:
To day I beleieve ORT doesn't take dependency on ONNX. If we add this, we are going to. Is that OK?

@spandantiwari @liqunfu

Reply:
@BowenBao How did you make this change suggestion on github? I had no idea this was possible :-D